packages/react/src/ReactSharedInternals.js

```diff
@@ -0,0 +1,28 @@
+/**
+ * Copyright (c) 2013-present, Facebook, Inc.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import assign from 'object-assign';
+import ReactCurrentOwner from './ReactCurrentOwner';
+import ReactDebugCurrentFrame from './ReactDebugCurrentFrame';
+
+const ReactSharedInternals = {
+  ReactCurrentOwner,
+  // Used by renderers to avoid bundling object-assign twice in UMD bundles:
+  assign,
+};
+
+if (__DEV__) {
+  Object.assign(ReactSharedInternals, {
```

Comment:
I know this code was just moved, but I'm curious why assign is imported above, but Object.assign is used here. Are they equivalent?

Reply:
We *could* put `Object.assign` there directly and assume it'll be compiled to an `assign` import but that would obscure the reason we have this code in the first place (which is to attach this specific npm dependency)

Reply:
In other words:

* We compile `Object.assign` calls to `assign` imports
* But we also replace `assign` with `React.__internals.assign` in renderer bundles to avoid bundling `assign` twice for UMD builds
* So we need to put `assign` on the React internals object

Reply:
I didn't realize Object.assign compiled down to the `assign` module, 👍

Reply:
Hm, it's early so maybe Dan's explanation makes sense– but after reading it twice, I'm still confused about why we import `assign`:
```js
import assign from 'object-assign';
```
But then use `Object.assign` here.

Reply:
>why we don't also just use assign here instead of Object.assign 😄

We could, I just didn't see the point in explicitly using it in a situation where we normally wouldn't.

Reply:
In other words — they're the same thing. The intention of the code is "put this Node module onto an object". So I'm using `assign` for the "Node module" part to express that intention, and use `Object.assign` for the "put onto an object" part because that's how we would express it in any other file. Otherwise you start thinking: "okay, this file uses `assign()` from npm instead of `Object.assign`, maybe I need to do this in other files too.."

Reply:
That...makes sense, but I still find the distinction (within this file) confusing. And I suspect others might too. 😄

Reply:
Open to other suggestions, I don’t really care either way. Feel free to change it to what’s looks less confusing.

This isn’t new code.

Reply:
I'm not really sure what's better, and I can go either way. I don't consider it a blocker for getting this change in.